src/main/scala/ru/d4team/chat/services/ChatRoomsService.scala

```diff
@@ -33,4 +34,12 @@ final case class ChatRoomsServiceMapImpl(private[services] val rooms: ChatRoomsM
         case None       => TrieMap(channel.id -> channel).some
       }
     }.unit
+
+  def leave(channel: Channel[WebSocketFrame], room: String): UIO[Unit] =
+    ZIO.succeed {
+      rooms.updateWith(room) {
+        case Some(room) => room.subtractOne(channel.id).some.filter(_.nonEmpty)
+        case None       => None // TODO (Aleksei Litkovetc) - Log a leaving from non-existent room
+      }
```

Comment:
Why don't we add this log now?

Reply:
It's not an effectful function, and we cannot add ZIO logging here (at least without unsafe run).
Adding a not effectful logger here is also not a good idea (we cannot test it properly and we will bring kind of overhead).
Another one option is to check count before and then proceed with updating or logging a case for non-existent room (but it would be not a thread safe).

Anyway, we can do all the things correct, but I would prefer to make it in another PR and keep current changes as simple as possible (and this approach matches well with trunk based development way).

Reply:
more than convincing)